repos/system_upgrade/el7toel8/actors/networkmanagerreadconfig/libraries/library.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from six.moves.configparser import ConfigParser, ParsingError
+
+from leapp.libraries.stdlib import CalledProcessError, run, api
+
+
+def _is_file_readable(path):
+    """ Verify if file exists and is readable """
+    return os.path.isfile(path) and os.access(path, os.R_OK)
+
+
+def read_nm_config(file_path=None):
+    if file_path:
+        try:
+            with open(file_path, 'r') as f:
+                r = f.read()
+                return r
+        except IOError:
+            return None
+    else:
+        try:
+            # Use 'NM --print-config' to read the configurationo so
+            # that the main configuration file and other files in
+            # various directories get merged in the right way.
+            r = run(['NetworkManager', '--print-config'], split=False)['stdout']
+            return r
+        except (OSError, CalledProcessError) as e:
+            api.current_logger().warning('Error reading NetworkManager configuration: {}'.format(e))
+            return None
+
+
+def parser_nm_config(cfg):
+    parser = ConfigParser()
+
+    try:
+        if hasattr(parser, 'read_string'):
+            # Python 3
+            parser.read_string(cfg)
+        else:
+            # Python 2
+            from cStringIO import StringIO
+            parser.readfp(StringIO(cfg))
```

Comment:
Thi will fail for `cfg = None`, which happens when `read_nm_config` returns `None`.

Reply:
I still don't think this is resolved.

Reply:
Yes, you are right, I missed the `except TypeError` clause.